### PR TITLE
chore!: remove deprecated items from `coder_workspace` data source

### PR DESCRIPTION
**Pull request by matifali**
The owner-related fields (`owner`, `owner_email`, `owner_id`, `owner_name`, `owner_oidc_access_token`, `owner_groups`, `owner_session_token`) in the workspace data source have been deprecated. This commit removes these fields from the codebase.

We should have done this before https://github.com/coder/terraform-provider-coder/releases/tag/v1.0.0

contributes to #250

### CODE REVIEW

File: docs/data-sources/workspace.md [hunk 1]
```diff
@@ -13,11 +13,51 @@ Use this data source to get information for the active workspace build.
 ## Example Usage
 
 ```terraform
```
Comment (matifali) [inline docs/data-sources/workspace.md:15]: Elaborated example usage. If its too much happy to simply it.

File: docs/data-sources/workspace_owner.md [hunk 2]
```diff
@@ -36,7 +34,7 @@ resource "coder_env" "git_author_name" {
 }
 
 resource "coder_env" "git_author_email" {
-  agent_id = var.agent_id
+  agent_id = coder_agent.dev.id
```
Comment (matifali) [inline docs/data-sources/workspace_owner.md:37]: there was no var defined so used the already defined agent.

File: integration/integration_test.go [hunk 3]
```diff
@@ -71,25 +71,18 @@ func TestIntegration(t *testing.T) {
...
-				"workspace.access_url":              `https?://\D+:\d+`,
-				"workspace.id":                      `[a-zA-z0-9-]+`,
-				"workspace.name":                    `test-data-source`,
-				"workspace.owner":                   `testing`,
-				"workspace.owner_email":             `testing@coder\.com`,
-				"workspace.owner_groups":            `\[\]`,
-				"workspace.owner_id":                `[a-zA-Z0-9]+`,
-				"workspace.owner_name":              `default`,
-				"workspace.owner_oidc_access_token": `^$`, // TODO: need a test OIDC integration
-				"workspace.owner_session_token":     `[a-zA-Z0-9-]+`,
```
Comment (matifali) [inline integration/integration_test.go:87]: removed all owner related items from the integration test. We already have a separate integration test for `coder_workspace_owner`.

**Review by matifali — COMMENTED** [review]
self review.

**Review by johnstcn — APPROVED** [review]

**Comment by johnstcn** [comment]
We will need to bump the major version after this change.

**Review by mtojek — APPROVED** [review]

**Pull request merged**